File: estudos/Coisas-Obscuras/Java/calculadora.java
import java.util.Scanner;

public class Calculadora {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        try {
            System.out.println("=== CALCULADORA ===");
            System.out.println("1 - Soma");
            System.out.println("2 - Subtração");
            System.out.println("3 - Multiplicação");
            System.out.println("4 - Divisão");

            System.out.print("Escolha: ");
            int opcao = scanner.nextInt();

            System.out.print("Primeiro número: ");
            double num1 = scanner.nextDouble();

            System.out.print("Segundo número: ");
            double num2 = scanner.nextDouble();

            double resultado = 0;

            switch (opcao) {
                case 1:
                    resultado = num1 + num2;
                    break;
                case 2:
                    resultado = num1 - num2;
                    break;
                case 3:
                    resultado = num1 * num2;
                    break;
                case 4:
                    if (num2 == 0) {
                        System.out.println("Não pode dividir por zero!");
                        return;
                    }
                    resultado = num1 / num2;
                    break;
                default:
                    System.out.println("Opção inválida!");
                    return;
            }

            System.out.println("Resultado: " + resultado);

        } catch (Exception e) {
            System.out.println("Erro: digite apenas números!");
        }

        scanner.close();
    }
}

File: estudos/Coisas-Obscuras/typescript/calculadora.ts
class Calculadora {
  somar(a: number, b: number): number {
    return a + b;
  }

  subtrair(a: number, b: number): number {
    return a - b;
  }

  multiplicar(a: number, b: number): number {
    return a * b;
  }

  dividir(a: number, b: number): number {
    if (b === 0) {
      throw new Error("Não é possível dividir por zero.");
    }
    return a / b;
  }
}

// Exemplo de uso
const calc = new Calculadora();

console.log("Soma:", calc.somar(10, 5));
console.log("Subtração:", calc.subtrair(10, 5));
console.log("Multiplicação:", calc.multiplicar(10, 5));
console.log("Divisão:", calc.dividir(10, 5));